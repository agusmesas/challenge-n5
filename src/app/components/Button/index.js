import styles from './style.module.scss';

const Button = ({ label, onClick, disabled, type = 'primary', children, className }) => {
  const handleOnClick = event => {
    event.preventDefault();
    event.stopPropagation();
    onClick();
  }

  return (
    <button
      className={`${styles.button} ${styles[`button--${type}`]} ${className}`}
      onClick={handleOnClick}
      disabled={disabled}
    >
      {label || children}
    </button>
  );
};

export default Button;