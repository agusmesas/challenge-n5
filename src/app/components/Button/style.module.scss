@import '../../utils/theme.scss';

.button {
  border: none;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-weight: bolder;
  margin-right: 10px;
  padding: 10px 15px;
  
  &--primary {
    background-color: $primary-color;
    color: $text-primary-color;

    &:hover {
      background-color: darken($color: $primary-color, $amount: 20%);
    }
  }

  &--secondary {
    background-color: $secondary-color;
    color: $text-secondary-color;

    &:hover {
      background-color: darken($color: $secondary-color, $amount: 50%);
    }
  }

  &:disabled {
    opacity: .2;
  }
}