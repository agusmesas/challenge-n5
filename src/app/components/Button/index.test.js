import '@testing-library/jest-dom';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from './';

const renderComponent = (propOverride) => {
  const props = {
    label: 'button text',
    onClick: jest.fn(),
    disabled: false,
    theme: 'primary',
    ...propOverride,
  }

  const view = render(<Button {...props} />);

  return { view, props };
}

describe('Button' , () => {
  it('should render correctly', async () => {
    const { view } = renderComponent();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should call onClick', async () => {
    const { props } = renderComponent();

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(props.onClick).toHaveBeenCalled();
  });
})