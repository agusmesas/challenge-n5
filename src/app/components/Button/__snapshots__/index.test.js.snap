// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<DocumentFragment>
  <button
    class="button button--primary "
  >
    button text
  </button>
</DocumentFragment>
`;
